src/Uno.UI/UI/Xaml/Controls/ToolTip/ToolTipService.cs

```diff
@@ -36,65 +42,91 @@ private static void OnToolTipChanged(DependencyObject dependencyobject, Dependen
 				return;
 			}
 
-			var toolTip = args.NewValue as ToolTip;
+			var newToolTip = args.NewValue as ToolTip;
+			var existingToolTip = GetToolTipObject(element);
 
-			if (toolTip == null && args.NewValue != null)
+			// Handle cases where the new value is not a ToolTip instance
+			if (newToolTip == null && args.NewValue != null)
 			{
-				toolTip = new ToolTip { Content = args.NewValue };
+				if (existingToolTip != null)
+				{
+					// Just replace content of the existing ToolTip.
+					existingToolTip.Content = args.NewValue;
+					return;
+				}
+				else
+				{
+					// We need to create a ToolTip object for the custom content.
+					newToolTip = new ToolTip { Content = args.NewValue };
+				}
 			}
 
-			if (toolTip != null)
+			// Remove existing ToolTip.
+			if (existingToolTip != null)
 			{
-				// First time: we're subscribing to event handlers
-
-				long currentHoverId = 0;
-
-				toolTip.SetAnchor(element);
+				existingToolTip.IsOpen = false;
+				SetToolTipObject(element, null);
+				element.PointerEntered -= OnPointerEntered;
+				element.PointerExited -= OnPointerExited;
 
-				element.Loaded += (snd, evt) =>
-				{
-					element.PointerEntered += OnPointerEntered;
-					element.PointerExited += OnPointerExited;
-				};
+				element.Unloaded -= OnOwnerUnloaded;
+			}
 
-				element.Unloaded += (snd, evt) =>
-				{
-					toolTip.IsOpen = false;
+			// Setup new ToolTip, if provided.
+			if (newToolTip != null)
+			{
+				newToolTip.SetAnchor(element);
+				SetToolTipObject(element, newToolTip);
 
-					element.PointerEntered -= OnPointerEntered;
-					element.PointerExited -= OnPointerExited;
-				};
+				element.PointerEntered += OnPointerEntered;
+				element.PointerExited += OnPointerExited;
 
-				void OnPointerEntered(object snd, PointerRoutedEventArgs evt)
-				{
-					var t = HoverTask(++currentHoverId);
-				}
+				element.Unloaded += OnOwnerUnloaded;
```

Comment:
I'm wondering if this will create leaks... You should try adding a control with a tooltip to the leaks unit test: 

https://github.com/unoplatform/uno/blob/56d3ba98ee6cab055dc8d7c4941ce4494e20e946/src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml/Given_FrameworkElement_And_Leak.cs#L64